### ISSUE TITLE
[ENH] `transformations` dependency set fails to install under windows due to `pycatch22 0.4.4`

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#5671]
The `transformations` dependency set fails to install under windows due to `pycatch22 0.4.4`.

This is a known failure, see here
https://github.com/DynamicsAndNeuralSystems/pycatch22/issues

that made its way into the 0.24.2 and 0.25.0 released due to `dependabot` only updating it in the `transformations` and not `all_extras` dependency sets, therefore eluding tests.

We should add the bound back (done in https://github.com/sktime/sktime/pull/5670) and release with the bound.

There is also a more general question in why `dependabot` updated it only in one place, and how we can ensure that the CI catches failures in all dependency sets.

FYI @yarnabrina

**Comment by yarnabrina** [comment]
> how we can ensure that the CI catches failures in all dependency sets

This was supposed to be covered by new CI itself, and #5375 was added there for this purpose. What it does is if there is a change in `pyproject.toml` in a PR, it tries to install all component-specific extras in all supported python versions in all supported operating systems.

However, the other changes added later, mainly the one to ensure code quality checks start first introduced at least 2 typos which I fixed in #5638. After the fixes, and verification in #5639, I think new CI will now be fully functional. So requesting review for those PR's.

**Comment by fkiraly** [comment]
Yes, thanks.

My question would be though, if we *deactivate* the old CI, the the (more importat) `all_extras` depedency set would not be checked?

**Comment by yarnabrina** [comment]
If you check (https://github.com/yarnabrina/sktime-fork/blob/f0ce0c6d1bff1a216da33137dfa3cd89310056b4/.github/workflows/test_components_without_extras.yml#L61-L62), all components for which I didn't create an extra for are tested with `all_extras_pandas2`.

Also, installation of `all_extras_pandas` is tested in all OS-python combinations (https://github.com/yarnabrina/sktime-fork/blob/f0ce0c6d1bff1a216da33137dfa3cd89310056b4/.github/workflows/validate_extras.yml#L24).

So, the way I see is that we'll only loose the coverage for components with extras with `all_extras_pandas2` installation, which is likely to install older versions than individual extras. Personally, this seems more realistic use case from user's perspective to me, as I don't know why will a user of sktime will attempt to install every single soft dependencies and not the specific ones (or specific extras) they need.

If this loss is a serious concern, we can run old CI (or a subset of it) in a scheduled fashion like I said in https://github.com/sktime/sktime/issues/5475#issuecomment-1862888007.

> So, I am hoping we can "retire" old CI in the form of running it every week/month/etc. (schedule + cron) in short term, and in long run disable it with option to trigger manually (workflow_dispatch) as and when needed.

---

After writing all this, it suddenly occurred to me that may be you have specifically asked for `all_extras` and not `all_extras_pandas2`? In that case, yes I'm not covering it at all (also not `pandas1`). Do you want to duplicate entirety of new CI with pandas v1?

**Comment by fkiraly** [comment]
> all components for which I didn't create an extra for are tested with

Yes, but is the following not correct then: the set up (new CI and old CI removed) would not catch the current issue reversed, i.e., if the bound would have been removed in `all_extras` but not in `transformations`.

In a sense this is "more severe", since afaik `all_extras` is a popular dep set.

**Comment by yarnabrina** [comment]
@5585 is an example of how dependabot is going to test installation of extras in case of change in pyproject.

Limitations as of now:

1. It tests everything as of now, and not just modified extras. Tracked in #5477.
2. It does not run unit tests (as part of new CI) for estimators which are affected by that dependency. Tracked in #5292.
